source/Assets/Plugins/Didomi/Tests/UITestsSuite.cs

```diff
@@ -0,0 +1,148 @@
+
+        yield return new WaitForSeconds(1);
+        Assert.True(Didomi.GetInstance().IsNoticeVisible(), "Notice should be visible");
+        CheckEvents("Called SetupUI", expectNoticeDisplayed: true);
+    }
+
+    [UnityTest]
+    public IEnumerator TestNoticeVisibility()
+    {
+        // On IOS, ShowNotice() requires a previous call to SetupUI()
+        yield return SetupUIWithoutNotice();
+
+        Didomi.GetInstance().ShowNotice();
+        yield return new WaitForSeconds(1);
+        Assert.True(Didomi.GetInstance().IsNoticeVisible(), "Notice should be visible");
+        CheckEvents("Called ShowNotice", expectNoticeDisplayed: true);
+
+        ResetEvents();
+
+        Didomi.GetInstance().HideNotice();
+        yield return new WaitForSeconds(1);
+        Assert.False(Didomi.GetInstance().IsNoticeVisible(), "Notice should not be visible anymore");
+        CheckEvents("Called HideNotice", expectNoticeHidden: true);
+    }
+
+    [UnityTest]
+    public IEnumerator TestPreferencesVisibility()
+    {
+        Didomi.GetInstance().ShowPreferences();
+        yield return new WaitForSeconds(1);
+        Assert.False(Didomi.GetInstance().IsNoticeVisible(), "Notice should not be visible");
+        Assert.True(Didomi.GetInstance().IsPreferencesVisible(), "Preferences screen should be visible");
+        CheckEvents("Called ShowPreferences", expectPreferencesDisplayed: true);
+
+        ResetEvents();
+
+        Didomi.GetInstance().HidePreferences();
+        yield return new WaitForSeconds(1);
+        Assert.False(Didomi.GetInstance().IsNoticeVisible(), "Notice should still not be visible");
+        Assert.False(Didomi.GetInstance().IsPreferencesVisible(), "Preferences screen should not be visible anymore");
+        CheckEvents("Called HidePreferences", expectPreferencesHidden: true);
+    }
+
+    private void EventListener_ShowNotice(object sender, ShowNoticeEvent e)
+    {
+        noticeDisplayedEvent = true;
+    }
+
+    private void EventListener_HideNotice(object sender, HideNoticeEvent e)
+    {
+        noticeHiddenEvent = true;
+    }
+
+    private void EventListener_ShowPreferences(object sender, ShowPreferencesEvent e)
+    {
+        preferencesDisplayedEvent = true;
+    }
+
+    private void EventListener_HidePreferences(object sender, HidePreferencesEvent e)
+    {
+        preferencesHiddenEvent = true;
+    }
+
+    private void ResetEvents()
+    {
+        noticeDisplayedEvent = false;
+        noticeHiddenEvent = false;
+        preferencesDisplayedEvent = false;
+        preferencesHiddenEvent = false;
+    }
+
+    private IEnumerator SetupUIWithoutNotice()
+    {
+        Didomi.GetInstance().SetupUI();
+        yield return new WaitForSeconds(1);
+        Didomi.GetInstance().HideNotice();
+        yield return new WaitForSeconds(1);
+        Assert.False(Didomi.GetInstance().IsNoticeVisible(), "Notice should not be visible at startup");
+        ResetEvents();
+    }
+
+    private void CheckEvents(
```

Comment:
minor suggestion: I recommend calling this method `AssertEvents` to be consistent with the calls that happen in it